cpp/src/decisiontree/quantile/quantile.cuh

```diff
@@ -183,5 +183,70 @@ void preprocess_quantile(const T *data, const unsigned int *rowids,
   return;
 }
 
+template <typename T>
+__global__ void computeQuantilesSorted(T *quantiles, const int n_bins,
+                                       const T *sorted_data, const int length) {
+  int tid = threadIdx.x + blockIdx.x * blockDim.x;
+  float bin_width = float(length) / n_bins;
```

Comment:
this should be: `  auto bin_width = static_cast<T>(length) / n_bins;`, right?

Reply:
The data type of `bin_width`  is not be dependent on data type of input data. It is actually dependent on length of the input data i.e. number of input samples and number of quantile bins. Having `bin_width` as `float` is fine in my opinion. If we are worried about floating point errors in `bin_width` due to large number of samples or bins, we should simply use `double` because even when input data is of type `float`, the array length could be large. 
Do you think we should switch to `double`?
Anyway I will change the type conversion to `static_cast`.

Reply:
good point. Agree with you. Then we should just switch to `double` here.